### PR TITLE
Issue #28 - moving mock SSH server creation into separate module

### PR DESCRIPTION
**Pull request by prondzyn**
I prepared a builder which provides unified place to prepare mocked SSH servers. Creating SSH servers in tests is now more readable. The builder is easily extendable for future functionality.

### CODE REVIEW

**Comment by dmarkov** [comment]
@prondzyn Thanks for your pull request, let me find someone who can review it

**Comment by dmarkov** [comment]
@dmzaytsev could you please review this pull request

**Comment by dmzaytsev** [comment]
@prondzyn generally we should try to avoid usage of utility classes
see http://www.yegor256.com/2014/05/05/oop-alternative-to-utility-classes.html#utility-classes-are-evil

**Comment by dmzaytsev** [comment]
@prondzyn according [our quality rules](http://www.teamed.io/qa.html) 
>Pull request description explains the solution proposed and contains a link to the original ticket it is related to.

please fix the description

**Comment by prondzyn** [comment]
@dmzaytsev I replaced the utility class with a more readable builder and fixed the description.

File: src/test/java/com/jcabi/ssh/SSHByPasswordTest.java [hunk 1]
```diff
@@ -29,40 +29,28 @@
...
 import org.hamcrest.Matchers;
 import org.junit.Test;
-import org.mockito.Mockito;
 
 /**
  * Unit tests for {@link SSHByPassword}.
- * @todo #21:30min SSH server setup in this class is nearly identical to the
- *  one in {@link SSHTest}. It should be extracted in a separate reusable
- *  module.
+ *
```
Comment (dmzaytsev) [inline src/test/java/com/jcabi/ssh/SSHByPasswordTest.java]: @prondzyn do we really need checkstyle suppression below ?
`@checkstyle ClassDataAbstractionCoupling (500 lines)`

File: src/test/java/com/jcabi/ssh/SSHByPasswordTest.java [hunk 2]
```diff
@@ -86,7 +74,8 @@
     @Test
     public void executesCommand() throws Exception {
         final int port = SSHByPasswordTest.port();
-        final SshServer sshd = SSHByPasswordTest.server(port);
+        final SshServer sshd = new MockSshServerBuilder(port)
+            .usePasswordAuthentication(LOGIN, PASSWORD).build();
```
Comment (dmzaytsev) [inline src/test/java/com/jcabi/ssh/SSHByPasswordTest.java]: @prondzyn variables `LOGIN` and `PASSWORD` can be local

**Comment by dmzaytsev** [comment]
@prondzyn thanks
please see two comments above

**Comment by prondzyn** [comment]
@dmzaytsev code cleaned according to your comments

**Comment by dmzaytsev** [comment]
@prondzyn thank you

**Comment by dmzaytsev** [comment]
@rultor merge

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by dmzaytsev** [comment]
@prondzyn please put the link to the original ticket into the pull request description

**Comment by prondzyn** [comment]
@dmzaytsev done

**Comment by dmarkov** [comment]
@dmzaytsev thanks, I just added **21 mins** to your account, payment `66377947`, 47 hours and 41 mins spent. extra minutes for review comments (c=6). +21 to your rating, your total score is [+1237](http://www.netbout.com/b/35166?open=rating)

**Comment by dmarkov** [comment]
@rultor please deploy

**Pull request merged**